todo/src/Components/TodoTaskForm.tsx

```diff
@@ -0,0 +1,127 @@
+import React from "react";
+import { Button, TextField } from "@mui/material";
+import { Grid } from "@mui/material";
+import { IFormData } from "../Todo";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText";
+import DialogTitle from "@mui/material/DialogTitle";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { makeStyles } from "@material-ui/styles";
+import TaskAction from "./TaskAction";
+
+const useStyles = makeStyles({
+  button: {
+    paddingTop: 30,
+    height: 40,
+  },
+  todoList: {
+    textAlign: "center",
+    justifyItems: "center",
+  },
+  input: {
+    height: 40,
+  },
+  span: {
+    color: "red",
+  },
+});
+
+const TodoTaskForm: React.FC = () => {
```

Comment:
Cant this same form be used for Task Edit too ?

Reply:
We can but I'm using separate components for add and edit.So,if I'm using edit in same form as add then it will be easy.
As I'm using separate components it will be little bit complex task to pass each and every time all parameters.

Reply:
So if have to add a new item to the form, then i will have to make changes at 2 places ?